### PR TITLE
chore: Introduce `ui-avatar` package to monorepo

### PR DESCRIPTION
**Pull request by tiagoevanp**
## Proposed changes (including videos or screenshots)

This PR introduces a new monorepo package for the `Avatar` component.

Reasons:
- Better maintainability for Avatar components
- Adoption of Rocket.Chat `Avatar` by another packages like `fuselage-ui-kit`

## Further comments

AECO-481

### CODE REVIEW

File: packages/ui-avatar/src/components/MessageAvatar.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const MessageAvatar: VFC<MessageAvatarProps> = ({ emoji, avatarUrl, username, size = 'x36', onClick, ...props }) => {
+	if (emoji) {
+		return (
+			<AvatarContainer size={size} {...props}>
+				{emoji}
+			</AvatarContainer>
+		);
+	}
+
+	return (
```
Comment (dougfabris) [inline packages/ui-avatar/src/components/MessageAvatar.tsx:24]: we could avoid this `Box`

File: packages/ui-avatar/src/components/MessageAvatar.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { AvatarContainer, Box } from '@rocket.chat/fuselage';
+import type { ReactElement, VFC } from 'react';
+
+import { UiAvatarProps } from './Avatar';
+import UserAvatar from './UserAvatar';
+
+type MessageAvatarProps = {
```
Comment (dougfabris) [inline packages/ui-avatar/src/components/MessageAvatar.tsx:7]: we could extend the proper props

File: packages/ui-avatar/src/components/RoomAvatar.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	url?: string;
+	room: {
+		_id: string;
+		type?: string;
+		t?: string;
+		avatarETag?: string;
+	};
+};
+
+const RoomAvatar = function RoomAvatar({ room, url, size }: RoomAvatarProps): ReactElement {
```
Comment (dougfabris) [inline packages/ui-avatar/src/components/RoomAvatar.tsx:17]: we could spread the props properly

File: packages/ui-avatar/src/components/UserAvatar.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import Avatar, { UiAvatarProps } from './Avatar';
+
+type UserAvatarProps = Pick<UiAvatarProps, 'size'> & {
+	username: string;
+	etag?: string;
+	url?: string;
+	title?: string;
+};
+
+const UserAvatar: FC<UserAvatarProps> = ({ username, etag, size }) => {
```
Comment (dougfabris) [inline packages/ui-avatar/src/components/UserAvatar.tsx:14]: we could spread the props properly

File: apps/meteor/client/views/meet/CallPage.tsx [hunk 5]
```diff
@@ -217,15 +217,14 @@ const CallPage: FC<CallPageProps> = ({
...
 						}}
 					></video>
-					<UserAvatar
+					<Box
 						style={{
 							display: isCameraOn ? 'none' : 'block',
 							margin: 'auto',
 						}}
-						username={localAvatar}
-						className='rcx-message__avatar'
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/meet/CallPage.tsx:226]: I was not able to find the class rules in any css file in this or fuselage repo. So I assume is unnecessary...

File: apps/meteor/client/views/meet/CallPage.tsx [hunk 6]
```diff
@@ -293,15 +292,15 @@ const CallPage: FC<CallPageProps> = ({
...
 					}}
 				>
-					<UserAvatar
+					<Box
 						style={{
 							display: 'block',
 							margin: 'auto',
 						}}
-						username={remoteAvatar}
-						className='rcx-message__avatar'
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/meet/CallPage.tsx:302]: I was not able to find the class rules in any css file in this or fuselage repo. So I assume is unnecessary...

File: apps/meteor/client/views/meet/CallPage.tsx [hunk 7]
```diff
@@ -357,15 +355,14 @@ const CallPage: FC<CallPageProps> = ({
...
 							alignItems='center'
 						>
-							<UserAvatar
+							<Box
 								style={{
 									display: 'block',
 									margin: 'auto',
 								}}
-								username={visitorName}
-								className='rcx-message__avatar'
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/meet/CallPage.tsx:366]: I was not able to find the class rules in any css file in this or fuselage repo. So I assume is unnecessary...

File: apps/meteor/client/views/meet/MeetPage.tsx [hunk 8]
```diff
@@ -90,15 +90,14 @@ const MeetPage = () => {
...
 						alignItems='center'
 					>
-						<UserAvatar
+						<Box
 							style={{
 								display: 'block',
 								margin: 'auto',
 							}}
-							username={visitorToken ? visitorName : agentName}
-							className='rcx-message__avatar'
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/meet/MeetPage.tsx:99]: I was not able to find the class rules in any css file in this or fuselage repo. So I assume is unnecessary...

File: apps/meteor/client/views/meet/MeetPage.tsx [hunk 9]
```diff
@@ -111,15 +110,14 @@ const MeetPage = () => {
...
 						alignItems='center'
 					>
-						<UserAvatar
+						<Box
 							style={{
 								display: 'block',
 								margin: 'auto',
 							}}
-							username={visitorToken ? agentName : visitorName}
-							className='rcx-message__avatar'
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/meet/MeetPage.tsx:120]: I was not able to find the class rules in any css file in this or fuselage repo. So I assume is unnecessary...

File: apps/meteor/client/views/omnichannel/contactHistory/ContactHistoryItem.tsx [hunk 10]
```diff
@@ -32,9 +32,7 @@ function ContactHistoryItem({ history, setChatId, ...props }: ContactHistoryItem
 
 	return (
 		<Box pbs={16} is={Message} onClick={onClick} data-qa='chat-history-item' {...props}>
-			<Message.LeftContainer>
-				{username && <UserAvatar username={username} className='rcx-message__avatar' size='x36' />}
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/omnichannel/contactHistory/ContactHistoryItem.tsx:36]: I was not able to find the class rules in any css file in this or fuselage repo. So I assume is unnecessary...

File: apps/meteor/client/views/room/contextualBar/VideoConference/VideoConfList/VideoConfListItem.tsx [hunk 11]
```diff
@@ -61,9 +61,7 @@ const VideoConfListItem = ({
 			pb={8}
 		>
 			<Message {...props}>
-				<Message.LeftContainer>
-					{username && <UserAvatar username={username} className='rcx-message__avatar' size='x36' />}
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/room/contextualBar/VideoConference/VideoConfList/VideoConfListItem.tsx:65]: I was not able to find the class rules in any css file in this or fuselage repo. So I assume is unnecessary...

File: packages/fuselage-ui-kit/package.json [hunk 12]
```diff
@@ -68,6 +68,7 @@
     "@rocket.chat/icons": "^0.33.0",
     "@rocket.chat/prettier-config": "~0.31.25",
     "@rocket.chat/styled": "~0.31.25",
+    "@rocket.chat/ui-avatar": "workspace:^",
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply packages/fuselage-ui-kit/package.json:71]: It is used only inside of a storybook file...

File: packages/fuselage-ui-kit/src/stories/Message.stories.tsx [hunk 13]
```diff
@@ -1,7 +1,9 @@
 /* eslint-disable new-cap */
-import { Message, Avatar } from '@rocket.chat/fuselage';
+import { Message } from '@rocket.chat/fuselage';
+import { Avatar } from '@rocket.chat/ui-avatar';
 import type * as UiKit from '@rocket.chat/ui-kit';
 import { action } from '@storybook/addon-actions';
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply packages/fuselage-ui-kit/src/stories/Message.stories.tsx:6]: The TS config from this package is complaining about the necessity of React module import... I'm not sure about which version we of React we are using here...

File: packages/fuselage-ui-kit/src/stories/Message.stories.tsx [hunk 14]
```diff
@@ -54,10 +56,6 @@ const createStory = (blocks: readonly UiKit.LayoutBlock[]) => {
             value={{
               action: action('action'),
               updateState: action('updateState'),
-              payload: {
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply packages/fuselage-ui-kit/src/stories/Message.stories.tsx:57]: There is no more payload necessity for this block...

File: packages/fuselage-ui-kit/src/stories/Message.stories.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (dougfabris) [reply packages/fuselage-ui-kit/src/stories/Message.stories.tsx:6]: yep, I had the same issue and already added as you did, we can desconsider this.

File: packages/fuselage-ui-kit/package.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (dougfabris) [reply packages/fuselage-ui-kit/package.json:71]: Don't we offer avatars in component fuselage-ui-kit components? 🤔

File: apps/meteor/client/components/avatar/UserAvatarEditor/UserAvatarEditor.tsx [hunk 15]
```diff
@@ -77,10 +77,10 @@ function UserAvatarEditor({ currentUsername, username, setAvatarObj, disabled, e
 					url={url}
 					username={currentUsername || ''}
 					etag={etag}
-					style={{
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/components/avatar/UserAvatarEditor/UserAvatarEditor.tsx:85]: ok, going back with style on it...

File: packages/ui-avatar/src/components/MessageAvatar.tsx (hunk 1 above)
Comment (tiagoevanp) [reply packages/ui-avatar/src/components/MessageAvatar.tsx:24]: The only component who need a onClick event is the MessageAvatar, why should I pass down to the base Avatar the unnecessary event?

File: packages/ui-video-conf/src/VideoConfMessage/VideoConfMessage.stories.tsx [hunk 16]
```diff
@@ -1,4 +1,5 @@
-import { MessageDivider, Message, Avatar, Box } from '@rocket.chat/fuselage';
+import { MessageDivider, Message, Box } from '@rocket.chat/fuselage';
+import { Avatar } from '@rocket.chat/ui-avatar';
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply packages/ui-video-conf/src/VideoConfMessage/VideoConfMessage.stories.tsx:2]: ok, going back!

File: packages/ui-video-conf/src/VideoConfMessage/VideoConfMessageUserStack.tsx [hunk 17]
```diff
@@ -1,5 +1,6 @@
 import type { IVideoConferenceUser, Serialized } from '@rocket.chat/core-typings';
-import { Avatar, Box } from '@rocket.chat/fuselage';
+import { Box } from '@rocket.chat/fuselage';
+import { Avatar } from '@rocket.chat/ui-avatar';
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply packages/ui-video-conf/src/VideoConfMessage/VideoConfMessageUserStack.tsx:3]: ok, going back!

File: packages/ui-video-conf/package.json [hunk 18]
```diff
@@ -53,7 +54,8 @@
 		"/dist"
 	],
 	"dependencies": {
-		"@rocket.chat/emitter": "~0.31.25"
+		"@rocket.chat/emitter": "~0.31.25",
+		"@rocket.chat/ui-avatar": "workspace:^"
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply packages/ui-video-conf/package.json:58]: It's added as peerDependency, not sure if this package could be used independent or not... WDYT?

File: packages/ui-video-conf/src/VideoConfPopup/VideoConfPopup.stories.tsx [hunk 19]
```diff
@@ -1,4 +1,5 @@
-import { Avatar, Icon } from '@rocket.chat/fuselage';
+import { Icon } from '@rocket.chat/fuselage';
+import { Avatar } from '@rocket.chat/ui-avatar';
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply packages/ui-video-conf/src/VideoConfPopup/VideoConfPopup.stories.tsx:2]: ok, going back!

File: packages/uikit-playground/src/Components/Preview/Display/Surface/ContextualBarSurface.tsx [hunk 20]
```diff
@@ -10,6 +9,7 @@ import {
 	ContextualbarTitle,
 	Margins,
 } from '@rocket.chat/fuselage';
+import { Avatar } from '@rocket.chat/ui-avatar';
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply packages/uikit-playground/src/Components/Preview/Display/Surface/ContextualBarSurface.tsx:11]: ok, going back!

File: packages/uikit-playground/src/Components/Preview/Display/Surface/MessageSurface.tsx [hunk 21]
```diff
@@ -1,13 +1,14 @@
-import { Message, Avatar } from '@rocket.chat/fuselage';
+import { Message } from '@rocket.chat/fuselage';
+import { Avatar } from '@rocket.chat/ui-avatar';
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply packages/uikit-playground/src/Components/Preview/Display/Surface/MessageSurface.tsx:2]: ok, going back!

File: packages/fuselage-ui-kit/src/stories/Message.stories.tsx [hunk 22]
```diff
@@ -1,7 +1,9 @@
 /* eslint-disable new-cap */
-import { Message, Avatar } from '@rocket.chat/fuselage';
+import { Message } from '@rocket.chat/fuselage';
+import { Avatar } from '@rocket.chat/ui-avatar';
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply packages/fuselage-ui-kit/src/stories/Message.stories.tsx:3]: ok, going back!

File: apps/meteor/ee/client/apps/gameCenter/GameCenterList.tsx [hunk 23]
```diff
@@ -1,4 +1,5 @@
-import { Avatar, Icon, Table, TableBody, TableCell, TableHead, TableRow } from '@rocket.chat/fuselage';
+import { Icon, Table, TableBody, TableCell, TableHead, TableRow } from '@rocket.chat/fuselage';
+import { Avatar } from '@rocket.chat/ui-avatar';
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/ee/client/apps/gameCenter/GameCenterList.tsx:2]: ok, going back!

File: apps/meteor/ee/client/apps/gameCenter/GameCenterContainer.tsx [hunk 24]
```diff
@@ -1,4 +1,4 @@
-import { Avatar } from '@rocket.chat/fuselage';
+import { Avatar } from '@rocket.chat/ui-avatar';
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/ee/client/apps/gameCenter/GameCenterContainer.tsx:1]: ok, going back!

File: apps/meteor/client/views/omnichannel/contactHistory/MessageList/ContactHistoryMessage.tsx
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/omnichannel/contactHistory/MessageList/ContactHistoryMessage.tsx:1]: I think there is no necessity to pass the event to Avatar itself. There are places who will use UserAvatar without this strictly necessity... It looks like a specific thing for a specific place usage...

**Review by dougfabris — APPROVED** [review]
we're good to good

**Review by MartinSchoeler — APPROVED** [review]

**Pull request merged**